test/gtest/ucs/test_memtrack.cc

```diff
@@ -263,3 +263,56 @@ UCS_TEST_F(test_memtrack, custom) {
 
     test_total(1, ALLOC_SIZE);
 }
+
+
+class test_memtrack_log : public ucs::test {
+public:
+    static ucs_log_func_rc_t
+    detect_zero_size_logger(const char *file, unsigned line,
+                            const char *function, ucs_log_level_t level,
+                            const ucs_log_component_config_t *comp_conf,
+                            const char *message, va_list ap)
+    {
+        if (level == UCS_LOG_LEVEL_DEBUG) {
+            std::string err_str = format_message(message, ap);
+            if (err_str.find("zero-size allocation") != std::string::npos) {
+                ++log_count;
+                return UCS_LOG_FUNC_RC_STOP;
+            }
+        }
+        return UCS_LOG_FUNC_RC_CONTINUE;
+    }
+
+    static size_t log_count;
+    static ucs_log_level_t orig_log_level;
+};
+
+size_t test_memtrack_log::log_count = 0;
+ucs_log_level_t test_memtrack_log::orig_log_level;
+
+UCS_TEST_F(test_memtrack_log, zero_size) {
+    orig_log_level = ucs_global_opts.log_component.log_level;
```

Comment:
maybe add option to set/restore log level to scoped_log_handler, as optional second parameter to the constructor? for example default param value is UCS_LOG_LEVEL_LAST which means do-nothing.

Reply:
Replaced `ucs_debug` by `ucs_warn`.